### ISSUE TITLE
Is thermite supposed to download and install rust and compile the extension?

### ISSUE THREAD

**Issue by NullVoxPopuli** [malept/thermite#16]
Here are the changes I'm trying to make: https://github.com/NullVoxPopuli/case_transform-rust-extensions/pull/6/files

I was lead on to believe that I wouldn't need to write a download and install script myself? maybe? 
via: https://github.com/danielpclark/faster_path/issues/89#issuecomment-258695232

Basically, I'd like rust to install and my extension to build as part of the bundle install process. (I have this working in my master branch of my extension, but I'm having a hard time getting thermite to work via the readme's instructions -- what am I missing)

**Comment by malept** [comment]
No, downloading and installing Rust is outside of the scope of Thermite. In the `faster_path` issue you linked to, I said:

> [Thermite] accomplishes this in a different way, by installing pre-compiled Rust extensions.

Thermite assumes one of two scenarios:

1. The user already has Rust installed (plus Ruby development headers, etc.)
2. The user does not have Rust installed, but has network access.

The first case runs just like a C extension would - building from source. In the second case, what happens next depends on how the project developer configured Thermite. If they set the `github_releases` option to `true`, then it will search the GitHub Releases of the related project for the appropriate tarball containing a pre-compiled Rust-based extension. Otherwise it will fail, since `cargo` was not found.

To populate GitHub Releases, I suggest using CI. The [`rusty_blank`](https://github.com/malept/rusty_blank) project has an example of this.

TL;DR: Thermite's philosophy is "You don't need to install Rust to install Rust-based Ruby extensions."

**Comment by NullVoxPopuli** [comment]
Cool,  are rust bins cross platform? As long as the processor architecture is the same?

**Comment by malept** [comment]
You need to create one tarball for each Ruby minor version + target platform + target architecture that you intend to support. For example: https://github.com/malept/rusty_blank/releases/tag/v0.1.12-rust
